### DISCUSSION TITLE
[BUG] New-PnPSiteGroup randomly fails with "Unknown Error"

### DISCUSSION THREAD

**Discussion by iloving** [pnp/powershell#2300] [General]
I have a script that creates and configures a new sharepoint site.  Part of the process creates multiple document libraries, and I create separate sharepoint groups to manage the permissions of each.

I do this via the following command:
`$group = New-PnPSiteGroup -Connection $Connection -Name "$groupTitle" -Site "$SiteFullURL" -PermissionLevels Read -ErrorAction Continue -ErrorVariable result`

It will randomly fail for no apparent reason:

New-PnPSiteGroup: /_$SCRIPTPATH_/MasterSiteCreateScript.ps1:414:22
Line |
 414 |  …    $group = New-PnPSiteGroup -Connection $Connection -Name "$groupTit …
     |                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Unknown Error

### Expected behavior 
I expect the group to be created

### Actual behavior 
I get the above error.

### Steps to reproduce behavior
I have no idea how to reproduce it.  It's completely random.  I have put it in a loop to retry the operation several times before giving up.  After several attempts, it eventually works.

### What is the version of the Cmdlet module you are running?
Confirmed in 1.10.0 and 1.11.0

### Which operating system/environment are you running PnP PowerShell on?
- [ ] Windows
- [ ] Linux
- [X] MacOS
- [ ] Azure Cloud Shell
- [ ] Azure Functions
- [ ] Other : please specify

**Comment by CallumCrowley** [comment]
Please set tracing to "debug" using Set-PnPTraceLog and provide further information. This will allow the community to better understand the issue.

**Comment by iloving** [comment]
Here is what I got in the log:
Creating group 'ilsatest delivery' <--my write-output
pwsh Error: 0 : 2022-08-29 11:34:32.6664        [PnP.Framework] [0]     [Error] ExecuteQuery threw following exception: Microsoft.SharePoint.Client.ServerException: Unknown Error
   at Microsoft.SharePoint.Client.ClientRequest.ProcessResponseStream(Stream responseStream)
   at Microsoft.SharePoint.Client.ClientRequest.ProcessResponse()
   at Microsoft.SharePoint.Client.ClientRequest.ExecuteQueryToServerAsync(ChunkStringBuilder sb)
   at Microsoft.SharePoint.Client.ClientRequest.ExecuteQueryAsync()
   at Microsoft.SharePoint.Client.ClientRuntimeContext.ExecuteQueryAsync()
   at Microsoft.SharePoint.Client.ClientContext.ExecuteQueryAsync()
   at Microsoft.SharePoint.Client.ClientContextExtensions.ExecuteQueryImplementation(ClientRuntimeContext clientContext, Int32 retryCount, String userAgent)
ServerErrorCode: -1
ServerErrorTypeName: Microsoft.SharePoint.Client.UnknownError
ServerErrorTraceCorrelationId: a4c65fa0-f0d6-2000-537d-24d7549fc01e
ServerErrorValue: 
ServerErrorDetails: 
.       0ms
New-PnPSiteGroup: $MYPATH/MasterSiteCreateScript.ps1:431:22
Line |
 431 |  …    $group = New-PnPSiteGroup -Connection $Connection -Name "$groupTit …
     |                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Unknown Error

**Comment by gautamdsheth** [comment]
Same here as well, will have to move this to discussions since this an issue with the environment itself and not the cmdlet.
